src/overlays/actors/ovl_En_Si/z_en_si.c

```diff
@@ -77,24 +86,80 @@ static DamageTable D_8098CDB8 = {
     /* Powder Keg     */ DMG_ENTRY(1, 0x0),
 };
 
-#endif
-
-extern ColliderSphereInit D_8098CD80;
-extern CollisionCheckInfoInit2 D_8098CDAC;
-extern DamageTable D_8098CDB8;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CA20.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CAD0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CB70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/func_8098CBDC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Si/EnSi_Draw.s")
+void func_8098CA20(EnSi* this, PlayState* play) {
+    this->colliderSphere.dim.worldSphere.center.x = this->actor.world.pos.x;
+    this->colliderSphere.dim.worldSphere.center.y = this->actor.world.pos.y;
+    this->colliderSphere.dim.worldSphere.center.z = this->actor.world.pos.z;
+    this->colliderSphere.dim.worldSphere.radius =
+        this->colliderSphere.dim.modelSphere.radius * this->colliderSphere.dim.scale;
+    if (this->actor.colChkInfo.health > 0) {
+        CollisionCheck_SetAC(play, &play->colChkCtx, &this->colliderSphere.base);
+    }
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->colliderSphere.base);
+}
+
+void func_8098CAD0(EnSi* this, PlayState* play) {
```

Comment:
Probably `EnSiGiveToken` will work for this (can include setting the flag too, but that seems more like a side effect than the main purpose to me).